vm_insnhelper.c

```diff
@@ -1111,8 +1111,11 @@ vm_getivar(VALUE obj, ID id, IVC ic, const struct rb_callcache *cc, int is_attr)
             return val;
         }
         else {
-            if (!is_attr && RTEST(ruby_verbose)) {
-                rb_warning("instance variable %"PRIsVALUE" not initialized", QUOTE_ID(id));
+            if (!is_attr) {
+                if (RTEST(ruby_verbose)) {
+                    rb_warning("instance variable %"PRIsVALUE" not initialized", QUOTE_ID(id));
+                }
+                EXEC_EVENT_HOOK(GET_EC(), RUBY_EVENT_WARNING, obj, id, rb_intern("uninitialized_ivar"), Qnil, Qnil);
```

Comment:
I think this event should only be called in verbose mode, just as the warning is only issued in verbose mode.  It doesn't make sense to emit an event for a warning if there is no warning.

Reply:
Maybe we should rename the event then?  As I said in the subject of this PR, the point is to give Did You Mean better information about missing instance variables.  If disabling warnings hobbles DYM, then I don't think the tracepoint is very useful.

Reply:
If we do this, we should rename the event at the least.  In that case, I am against this feature as it would slow down use of uninitialized instance variables.

If Rails wants to enable these DYM warnings, can't it turn verbose mode on by default?  With Ruby 2.4+, Rails could use `Warning.warn` to filter any verbose warnings it doesn't want to raise to the user.  I think Rails already only supports Ruby 2.5+, so you should be able to rely on that.

Reply:
> In that case, I am against this feature as it would slow down use of uninitialized instance variables.

Maybe we should add a benchmark?  In a real program I don't think this is hotter than [RUBY_EVENT_RETURN](https://github.com/ruby/ruby/blob/65c5a3957877e67bae3bdbf063aac6f257a52ac8/vm.c#L1084) (or call) for example.



> If Rails wants to enable these DYM warnings, can't it turn verbose mode on by default? With Ruby 2.4+, Rails could use `Warning.warn` to filter any verbose warnings it doesn't want to raise to the user. I think Rails already only supports Ruby 2.5+, so you should be able to rely on that.

Good idea, I like that.  I'll close this PR!  Thank you!

Reply:
I posted similar thoughts on Redmine.

Reply:
At least the TracePoint doesn't create the warning message string, which is expensive.
But still, if DidYouMean uses this by default, then most Ruby programs will, and I expect the overhead to be significant as the TracePoint would be used (`:return` is rarely used OTOH).

Reply:
Ya, it makes sense.  Though I still think that if we care about the performance of uninitialized ivars we should add benchmarks for them.